src/lib/log.c

```diff
@@ -38,14 +38,22 @@ static void log_internal(FILE *out, const char *file, int line, const char *labe
 	struct tm *timeinfo;
 	int printed;
 
-	/* if the message starts with '\n' add a line break before the message */
+	/* if the message starts with '\n' add a line break before the message
+	 * unless we are printing in debug mode, or to a file because that would
+	 * mess up the output, if that is the case remove those characters */
 	while (strncmp(format, "\n", 1) == 0 || strncmp(format, "\r", 1) == 0) {
-		if (out == stdout || out == stderr) {
+		if ((out == stdout || out == stderr) && (cur_log_level != LOG_DEBUG)) {
```

Comment:
shouldn't we print '\n' even if we are using --debug?

Reply:
the only command that is printing \r now is the progress report. I'd change it to print \n instead if debug is used.
And stop handling \r here.

Reply:
> shouldn't we print '\n' even if we are using --debug?

What I was thinking is that the "\n" at the beginning of the lines are mostly used to space messages between each other to improve the looks of the output. When using --debug you are printing more information, so the output doesn't really need that extra spacing and printing line break would result in debug messages that look like this:
```
2019-07-23 17:46:27 (src/curl.c:570)              Debug: some debug message
2019-07-23 17:46:27 (src/curl.c:438)              Debug: more messages
2019-07-23 17:46:27 (src/curl.c:579)              
2019-07-23 17:46:27 (src/bundle.c:777)            Bla bla bla
```
But maybe it is ok to print those empty lines even in debug mode. What do you think?

Reply:
> the only command that is printing \r now is the progress report. I'd change it to print \n instead if debug is used.
> And stop handling \r here.

The problem with handling it there instead of here is that here it would allow for more flexibility. I like the idea of being able to use \n and \r with the logger if necessary so you can format your output as you please. It is not very common that you would need to use the \r thing, but if you do, it would be supported.

Reply:
I think we should always print something if the user of that lib calls the print function. For example if the user calls info("\n") I prefer to print an empty line than not printing anything. If we are debugging the output looking for a bug we are going to see prints in code that aren't printed on the output. And that makes things harder. Imagine one output like

    2019-07-23 17:46:27 (src/curl.c:570)              Debug: some debug message
    2019-07-23 17:46:27 (src/curl.c:438)              Debug: more messages
    2019-07-23 17:46:27 (src/curl.c:579)              
    Segmentation fault

We know that segfault occurred after that call and not before.